syntax = "proto2";

option java_multiple_files = true;
option java_package = "com.github.tmarsteel.audionetwork.protocol";

message AudioReceiverAnnouncement {
	required uint32 magic_word = 1;
	required uint64 mac_address = 2;
	required string device_name = 4;
	required bool currently_streaming = 5;
}

message AudioData {
	required uint32 sample_rate = 1;
	required uint32 bytes_per_sample = 2;
	required bytes samples = 3;
}